metricbeat/module/apache/status/status.go

```diff
@@ -59,30 +56,33 @@ func (m *MetricSeter) Setup(ms *helper.MetricSet) error {
 		Password:         "",
 	}
 
-	if err := ms.Module.ProcessConfig(&config); err != nil {
-		return err
+	if err := base.Module().UnpackConfig(&config); err != nil {
+		return nil, err
 	}
 
-	m.URLs = make(map[string]string, len(ms.Config.Hosts))
+	metricSet := &MetricSet{
+		BaseMetricSet: base,
+		URLs:          make(map[string]string, len(base.Module().Config().Hosts)),
+	}
 
 	// Parse the config, create URLs, and check for errors.
 	var errs multierror.Errors
-	for _, host := range ms.Config.Hosts {
+	for _, host := range base.Module().Config().Hosts {
 		u, err := getURL(config.Username, config.Password, config.ServerStatusPath, host)
 		if err != nil {
 			errs = append(errs, err)
 			continue
 		}
-		m.URLs[host] = u.String()
+		metricSet.URLs[host] = u.String()
 		debugf("apache-status URL=%s", redactPassword(*u))
 	}
 
-	return errs.Err()
+	return metricSet, errs.Err()
 }
 
 // Fetch makes an HTTP request to fetch status metrics from the mod_status
 // endpoint.
-func (m *MetricSeter) Fetch(ms *helper.MetricSet, host string) (event common.MapStr, err error) {
+func (m *MetricSet) Fetch(host string) (common.MapStr, error) {
```

Comment:
Is the idea above that the base is part of the struct, so it is still accessible?

Side note: I had event and err defined more for convenience, so people "must" use these names. Works also without.